PWGHF/HFC/Tasks/taskCorrelationDsHadrons.cxx

```diff
@@ -273,6 +284,51 @@ struct HfTaskCorrelationDsHadrons {
     return (ptBinD != -1 && bdtScorePrompt >= mlOutputPromptMin->at(ptBinD) && bdtScorePrompt <= mlOutputPromptMax->at(ptBinD) && bdtScoreBkg <= mlOutputBkg->at(ptBinD));
   }
 
+  template <typename T1, typename EfficiencyContainer, typename Histogram, typename T2>
+  double getEfficiencyWeight(float ptD,
+                             const T1* binsPtEfficiencyD,
+                             const EfficiencyContainer* efficiencyD,
+                             Histogram* mEfficiencyD,
+                             const std::optional<float> ptAssoc = std::nullopt,
+                             const T2* binsPtEfficiencyAssoc = nullptr,
+                             const EfficiencyContainer* efficiencyAssoc = nullptr,
+                             Histogram* mEfficiencyAssoc = nullptr)
+  {
+    if (!applyEfficiency) {
+      return 1.;
+    }
+
+    double weight = 1.;
+    if (loadAccXEffFromCCDB) {
+      if (mEfficiencyD->GetBinContent(mEfficiencyD->FindBin(ptD)) != 0) {
+        weight = 1. / mEfficiencyD->GetBinContent(mEfficiencyD->FindBin(ptD));
+      } else {
+        LOG(fatal) << "A bin content in Ds-meson efficiency histogram is zero!";
+      }
```

Comment:
Just define a safe epsilon, e.g. `1.e-5` and throw a fatal if the efficiency is  `< epsilon`.
There is no need to have `else`.